File: solutions/erlang/triangle/1/src/triangle.erl
-module(triangle).

-export([kind/3]).


kind(A, B, C) when A+B < C; A+C < B; B+C < A -> 
  { error , "side lengths violate triangle inequality" };
kind(A, A, A) when A > 0 -> equilateral;
kind(A, A, A) -> { error , "all side lengths must be positive" };
kind(_, A, A) -> isosceles;
kind(A, A, _) -> isosceles;
kind(A, _, A) -> isosceles;
kind(_A, _B, _C) -> scalene.

File: solutions/python/triangle/1/triangle.py
def equilateral(sides):
    return is_triangle(sides) and sides[0] == sides[1] == sides[2]


def isosceles(sides):
    return is_triangle(sides) and (sides[0] == sides[1] or sides[1] == sides[2] or sides[2] == sides[0])


def scalene(sides):
    return is_triangle(sides) and sides[0] != sides[1] != sides[2] != sides[0]


def is_triangle(sides):
    a, b, c = sides
    return False if a + b < c or a + c < b or b + c < a or 0 == a == b == c else True 